util/cbmem/cbmem.c

```diff
@@ -22,12 +22,18 @@
 #include <commonlib/tcpa_log_serialized.h>
 #include <commonlib/coreboot_tables.h>
 
+#include <byteswap.h>
+
 #ifdef __OpenBSD__
 #include <sys/param.h>
 #include <sys/sysctl.h>
 #endif
 
 #define ARRAY_SIZE(a) (sizeof(a) / sizeof((a)[0]))
+#define FIX_ENDIANNESS(a) (sizeof(a) == 1 ? (a) :\
```

Comment:
This is definitely not a proper way of doing it. We cannot assume that FW endianness will always be different than OS, such approach will break on all x86. Even on PPC64 we could possibly have BE OS, so we have to know when to swap bytes and when not. This needs some redesign of coreboot tables because as of now there is no indication about endianness saved in them.

There are some functions that can convert from BE to host, like `ntohl()` used in `dt_update_cells()`, they can be used instead of `bswap`.

Reply:
Agree, this cannot be finally implemented like this. I have created a macro to make it easier to fix later making necessary checks in just one place instead of many.

Reply:
> This needs some redesign of coreboot tables because as of now there is no indication about endianness saved in them.

Looking at `sys/arch` Power seems to be the first non-LE architecture, so one way of doing it could be to always use LE for tables. Another way is to use a heuristic like PR already does: `header_bytes` and `header_checksum` in wrong endianness are easily detectable. In both cases compatibility is preserved.

"LBIO" could also be reversed if structure uses BE. And if it will always LE so far, size can be made 1 or 2 bytes and the rest would become flags to indicate endianness and something else in the future.